python/lsst/rubintv/production/allSky.py

```diff
@@ -0,0 +1,468 @@
+
+    def hasDayRolledOver(self):
+        """Check if the dayObs has rolled over.
+
+        Checks if the current dayObs is the one the class was instantiated with
+        and returns False if it is.
+        """
+        currentDay = getCurrentDayObs_int()
+        if currentDay == self.dayObsInt:
+            return False
+        elif currentDay == self.dayObsInt+1:
+            return True
+        else:
+            raise RuntimeError(f"Encountered non-linear time! Day animation was started with {self.dayObsInt}"
+                               f"and now current dayObs is {currentDay}!")
+
+    def _getConvertedFilename(self, filename):
+        """Get the filename and path to write the converted images to.
+        """
+        return os.path.join(self.outputImageDir, os.path.basename(filename))
+
+    def convertFiles(self, files, forceRegen=False):
+        """Convert a list of files using convertJpgScale(), writing the
+        converted files to self.outputImageDir
+
+        Parameters
+        ----------
+        files : `Iterable` of `str`
+            The set of files to convert
+        forceRegen : `bool`
+            Recreate the files even is they exist?
+
+        Returns
+        -------
+        files : `set`
+            The files which were converted.
+        """
+        convertedFiles = set()
+        for file in sorted(files):  # sort just helps debug
+            outputFilename = self._getConvertedFilename(file)
+            self.log.debug(f"Converting {file} to {outputFilename}")
+            if not DRY_RUN:
+                # XXX add a try to this? Does convert ever fail?
+                if os.path.exists(outputFilename):
+                    self.log.warning(f"Found already converted {outputFilename}")
+                    if forceRegen:
+                        convertJpgScale(file, outputFilename)
+                else:
+                    convertJpgScale(file, outputFilename)
+            convertedFiles.add(file)
+        return set(convertedFiles)
+
+    def animateFilesAndUpload(self, isFinal=True):
+        """Animate all the files in self.outputImageDir and upload to GCS.
+
+        If isFinal is False the filename will end with largest input seqNum in
+        the animation. If isFinal is True then it will end with seqNum_final.
+        """
+        files = sorted(getFilesetFromDir(self.outputImageDir))
+        lastfile = files[-1]
+        if isFinal:
+            seqNumStr = 'final'
+        else:
+            seqNum = seqNumFromFilename(lastfile)
+            seqNumStr = f"{seqNum:05}"
+
+        channel = 'all_sky_movies'
+        fakeDataId = {'day_obs': self.dayObsInt, 'seq_num': seqNumStr}
+        uploadAsFilename = _dataIdToFilename(channel, fakeDataId, extension='.mp4')
+        creationFilename = os.path.join(self.outputMovieDir, uploadAsFilename)
+        self.log.info(f"Creating movie from {self.outputImageDir} as {creationFilename}...")
+        if not DRY_RUN:
```

Comment:
I think this is missing the `HAS_GOOGLE` check that I commented about above.

Reply:
That's OK, the whole init will fail elsewhere if that's not available, so I've removed all checks as they're done on instantiating the `Uploader`